### ISSUE TITLE
Nested states that transition to themselves look weird

### ISSUE THREAD

**Issue by LevelbossMike** [sverweij/state-machine-cat#28]
Thanks for this great project and sorry for the strange title. When nested states reference themself the rendered outpout looks weird and it's hard to see what is happening in the statechart:

<img width="685" alt="bildschirmfoto 2018-10-18 um 19 17 40" src="https://user-images.githubusercontent.com/242299/47172209-1d088a00-d30b-11e8-94ed-14a30869a633.png">

### Expected Behavior
I would expect that a transition from a nested state to itself looks the same as a non nested-state to itself.

### Current Behavior
The transition of a nested state to itself is drawn inside of the nested state which is easy to miss when looking at the diagram.

### Possible Solution
Not sure 

### Steps to Reproduce (for bugs)
This is the diagram I am drawing:
```
initial,
"media player off",

"media player on" {
  stopped, playing, paused;

  stopped => playing : play;
  playing => stopped : stop;
  playing => paused  : pause;
  paused  => playing : pause;
  paused  => stopped : stop;
};

initial            => "media player off";
"media player off" => stopped           : power;
"media player on"  => "media player off" : power;
"media player off" => "media player off";
"media player on" => "media player on";

```
### Context
I'm trying to build a visualizer for an ember-addon https://github.com/LevelbossMike/ember-statecharts that I am working on and would like to use state-machine-cat as the project that creates the visual output. 

### Your Environment
I'm using the web-editor.

Thanks again for this great project! I'm using your diagramming tool nearly every day and it makes my dev-life much easier 🍻

**Comment by sverweij** [comment]
@LevelbossMike thanks for raising this issue, and yes, that needs to be improved. (also the self-transitions from non-nested entities have become a bit too tiny for my taste since the last major change). I'll try to tweak the output into submission.

Your add-on looks cool and very useful 👍 - wish something like this would exist for angular...

(And thanks for the compliments 😊 )

**Comment by sverweij** [comment]
Intermediate update: below you see the current situation and what I came up with after some graphviz trickery:

```smcat
ns {
    s => s: hi;
};

ns => ns: hi;
```
![before](https://user-images.githubusercontent.com/4822597/47255686-4fce9180-d475-11e8-887d-0a8967b008a5.png) ![after](https://user-images.githubusercontent.com/4822597/47255744-f024b600-d475-11e8-8995-4565240c4eb8.png)

**Comment by LevelbossMike** [comment]
this looks great! Is this already released? I still see the old behavior in the web-editor

<img width="921" alt="bildschirmfoto 2018-10-22 um 09 50 08" src="https://user-images.githubusercontent.com/242299/47282535-f2a31f00-d5df-11e8-8179-fecb7ebcedf1.png">

**Comment by namibj** [comment]
I found a solution for this issue, but it seems to only be applicable to fdp.
You need to use `splines=compound` instead of `splines=true`,
and the state transitions/edges have to start/end on the `"cluster_`<name of clustered state>`"` node  instead of the invisible helper node inside the cluster, as is currently done. The ` lhead="cluster_`<name of clustered state>`"` inside the brackets has to/should be removed.
It might be necessary to also remove `rankdir=LR` and add `overlap=scale`, at least those seem to have solved it for me.

**Comment by sverweij** [comment]
Thanks @namibj! - I'll tweak the `fdp` output specifically to include you recommendations in a separate PR. (I wish we could do the same for `dot`!)

For the `dot` (and probably the `osage`) algorithm I'm afraid we're stuck with another helper node to guide the self transition outside the clustered node...

**Comment by sverweij** [comment]
@LevelbossMike I haven't released it yet - have to do some juggling in the `dot` reporter to get to the result. I'll let you know in this thread when I have something testable ready.

**Comment by sverweij** [comment]
hi @LevelbossMike - a ping as promised: the PR above tries to fix the issue. I've published a first beta version:
- on npm (as state-machine-cat@4.1.5-beta-1 in the beta channel) 
- on https://state-machine-cat.js.org/dev

I've already found two kinks to iron out (see the PR - @namibj already pointed out one of em) - but am curious if this  works well enough for your scenario(s).

**Comment by sverweij** [comment]
Update: just deployed a fix for the first of the two kinks (with direction `left-right` and `right-left` self transitions to composite states look sorta ok now as well) and published them on npm (state-machine-cat@4.1.5-beta-2) and on https://state-machine-cat.js.org/dev.

**Comment by LevelbossMike** [comment]
@sverweij looks great :beers: 👍

**Comment by sverweij** [comment]
@LevelbossMike I've merged the PR, released to npm as 4.1.5 and to the main branch on state-machine-cat.js.org.

Thanks again for raising the issue and for the prompt feedback!
